composer/trainer/trainer.py

```diff
@@ -233,20 +225,12 @@ def __init__(
             grad_accum=grad_accum,
             precision=precision,
             precision_context=precision_context,
-            train_dataloader=DDPDataLoader(train_dataloader_spec.dataloader),
+            train_dataloader=DDPDataLoader(train_dataloader),
             eval_dataloader=DDPDataLoader(eval_dataloader_spec.dataloader),
         )
 
         # Steps per epoch
-        if train_subset_num_batches is not None:
-            if train_subset_num_batches > self.state.steps_per_epoch:
-                warnings.warn(
-                    textwrap.dedent(
-                        f"""SubsetNumBatchesWarning: The train_subset_num_batches({train_subset_num_batches})
-                        is greater than the number of batches in the training dataloader
-                        ({self.state.steps_per_epoch})"""))
-            else:
-                self.state.steps_per_epoch = train_subset_num_batches
+        self.state.steps_per_epoch = train_subset_num_batches
```

Comment:
curious to understand why train_subset and eval_subset are now handled differently?